uniffi_bindgen/src/templates/CallbackInterfaceTemplate.rs

```diff
@@ -0,0 +1,102 @@
+{#
+// For each callback interface in the UDL, we assume the caller has provided a corresponding
+// Rust `trait` with the declared methods. We provide the traits for sending it across the FFI.
+// If the caller's trait does not match the shape and types declared in the UDL then the rust
+// compiler will complain with a type error.
+// 
+// The generated proxy will implement `Drop`, `Send` and `Debug`.
+#}
+{% let trait_name = obj.name() -%}
+{% let trait_impl = format!("{}Proxy", trait_name) -%}
+{% let foreign_callback_holder = format!("foreign_callback_{}_holder", trait_name)|upper -%}
+
+// Register a foreign callback for getting across the FFI.
+static {{ foreign_callback_holder }}: std::sync::atomic::AtomicUsize = std::sync::atomic::AtomicUsize::new(0);
+
+#[no_mangle]
+pub extern "C" fn {{ obj.ffi_init_callback().name() }}(callback: uniffi::ForeignCallback) {
+    uniffi::set_foreign_callback(&{{ foreign_callback_holder }}, callback);
+}
+
+// Make an implementation which will shell out to the foreign language.
+#[derive(Debug)]
+struct {{ trait_impl }} {
+  handle: u64
+}
+
+// We need Send so we can stash the callbacks in objects we're sharing with foreign languages
+// i.e. in the handle map.
+// Prepared for our target trait to declare:
+//  `trait {{ trait_name }}: Send + std::fmt::Debug`
+unsafe impl Send for {{ trait_impl }} {}
```

Comment:
We should document here the reasons why it's safe to `impl Send` for this struct. Actually I'm surprised that it's not automatically `Send`, but I haven't finished following all the threads here yet.

Reply:
I tried commenting out this `unsafe impl Send` and `cargo test` still ran successfully to completion, which suggests to me that Rust is now automatically deriving a `Send` impl for us. Maybe a refactor has made it automatically `Send`?

What do you think about switching from `unsafe impl Send` to statically asserting that it is implemented? This would help remind us to re-check any assumptions here if we ever change the struct in a way that makes it no longer automatically `Send`.

(I'd also support taking a depending on the [`static_assertions`](https://docs.rs/static_assertions) crate for this, as it may be a bit awkward to express this as an expression suitable for use with `ffi_support::static_assert!`).